### PR TITLE
dopamine: init at 3.0.0-preview.26

### PR DESCRIPTION
**Pull request by Guanran928**
## Description of changes

The audio player that keeps it simple 
https://github.com/digimezzo/dopamine

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lelgenio** [comment]
Working fine

---

Result of `nixpkgs-review pr 315465` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dopamine</li>
  </ul>
</details>

File: pkgs/by-name/do/dopamine/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    '';
+
+  meta = {
+    changelog = "https://github.com/digimezzo/dopamine/blob/${version}/CHANGELOG.md";
+    description = "The audio player that keeps it simple";
+    homepage = "https://github.com/digimezzo/dopamine";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "dopamine";
+    maintainers = with lib.maintainers; [ Guanran928 ];
+    platforms = lib.platforms.all;
```
Comment (lelgenio) [inline pkgs/by-name/do/dopamine/package.nix:33]:
```suggestion
    platforms = [ "x86_64-linux" ];
```

The Appimage only works in `x86_64-linux`, right?

File: pkgs/by-name/do/dopamine/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  lib,
+  fetchurl,
+  appimageTools,
+}:
+appimageTools.wrapType2 rec {
+  pname = "dopamine";
+  version = "3.0.0-preview.25";
+
+  src = fetchurl {
+    url = "https://github.com/digimezzo/dopamine/releases/download/v3.0.0-preview.25/Dopamine-3.0.0-preview.25.AppImage";
```
Comment (SuperSandro2000) [inline pkgs/by-name/do/dopamine/package.nix:11]:
```suggestion
    url = "https://github.com/digimezzo/dopamine/releases/download/v${version}/Dopamine-${version}.AppImage";
```

File: pkgs/by-name/do/dopamine/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  extraInstallCommands =
+    let
+      contents = appimageTools.extract { inherit pname version src; };
+    in
+    ''
+      install -Dm644 ${contents}/dopamine.desktop $out/share/applications/dopamine.desktop
+      substituteInPlace $out/share/applications/dopamine.desktop \
+        --replace 'Exec=AppRun' 'Exec=${meta.mainProgram}'
```
Comment (SuperSandro2000) [inline pkgs/by-name/do/dopamine/package.nix:22]:
```suggestion
        --replace-fail 'Exec=AppRun' 'Exec=dopamine'
```

**Review by SuperSandro2000 — DISMISSED** [review]

**Comment by pinpox** [comment]
Result of `nixpkgs-review pr 315465` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dopamine</li>
  </ul>
</details>

**Review by pinpox — APPROVED** [review]

**Pull request merged**